### DISCUSSION TITLE
ui.run_with: Inherit parent app's docs/redoc/openapi.json settings

### DISCUSSION THREAD

**Discussion by ProbablyBrian** [zauberzeug/nicegui#1746] [Ideas / Feature Requests]
### Description

Take this example:

```python
from fastapi import FastAPI

app = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)


@app.get("/")
def home():
    return "This is my homepage"

```

This properly hides the /docs, /redoc, and /openapi.json schema pages when running the app.  Using this same configuration within ui.run_with places nicegui's docs and API schema at the default /docs, /redoc, and /openapi.json locations.

```python
from fastapi import FastAPI
from nicegui import ui

app = FastAPI(docs_url=None, redoc_url=None, openapi_url=None)


@app.get("/")
def home():
    return "This is my homepage"


ui.run_with(app)
```

I'd like to have a bit more control over where and what docs are displayed to either modify or hide these settings for the entire application (including those provided by the mounted NiceGUI app).

Is there a way to do this currently?

**Comment by falkoschindler** [comment]
Hi @ProbablyBrian!

> Is there a way to do this currently?

Not that I'm aware of. But I haven't worked with FastAPI's docs pages lately. Maybe there is a way to workaround this issue.

Apart from that, I wonder how this relates to #1188 and #1196. Does #1196 only solve part of the problem because it allows controlling the OpenAPI routes, but not docs and redoc?
